doozer/doozerlib/cli/scan_osh.py

```diff
@@ -0,0 +1,157 @@
+import click
+import koji
+from doozerlib.cli import cli, click_coroutine, pass_runtime
+from doozerlib.runtime import Runtime
+from doozerlib.exectools import fire_and_forget, cmd_gather
+from doozerlib.util import cprint
+from typing import Optional
+
+
+class ScanOshCli:
+    def __init__(self, runtime: Runtime, last_brew_event: int, dry_run: bool, nvrs: Optional[list],
+                 check_triggered: Optional[bool]):
+        self.runtime = runtime
+        self.last_brew_event = last_brew_event
+        self.dry_run = dry_run
+        self.brew_tags = []
+        self.specific_nvrs = nvrs
+        self.check_triggered = check_triggered
+        self.nvrs = []
+
+        # Initialize runtime and brewhub session
+        self.runtime.initialize(clone_distgits=False)
+        self.koji_session = koji.ClientSession(self.runtime.group_config.urls.brewhub)
+
+    async def untriggered_nvrs(self):
+        """
+        So we might have already triggered a scan for the same NVR. If this flag is enabled, we want to check
+        if it has and not re-trigger it again.
+        But please note that this will take a lot of time since it has to run for all 200+ packages.
+        """
+        untriggered_nvrs = []
+        for nvr in self.nvrs:
+            rc, _, _ = cmd_gather(f"osh-cli find-tasks --nvr {nvr}")
```

Comment:
We can create couroutines here to make this faster. If not in this PR, let's remove the async from method def

Reply:
Also consider using `exectools.limit_concurrency` decorator if doing so, we might not spawn all tasks at the same time.

Reply:
removing from PR for now https://github.com/openshift-eng/art-tools/pull/51/commits/7c715b307aa02efa5d521ca0e931c9a2ae18766e